### ISSUE TITLE
select on swiper slide cannot be clicked via mouse

### ISSUE THREAD

**Issue by l1229164** [nolimits4web/swiper#3862]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* Swiper Version: EXACT RELEASE VERSION OR COMMIT HASH, HERE.
* Platform/Target and Browser Versions: PLATFORM CLIENT YOU ARE TARGETING SUCH AS macOS, Windows, CORDOVA, IOS, ANDROID, CHROME, ETC.
* Live Link or JSFiddle/Codepen or website with isssue: PREFERABLY _(IF YOU WANT YOUR ISSUE TO BE RESOLVED ASAP)_.

### What you did
EXPLAIN WHAT YOU DID, PREFERABLY WITH CODE EXAMPLES, HERE.

### Expected Behavior
EXPLAIN WHAT IS TO BE EXPECTED, HERE.

### Actual Behavior
When I put a select element in swiperSlide it doesn't work for left key of mouse clicking,but right still ok

P.S. Remember, an issue is not the place to ask questions. You can use [Stack Overflow](http://stackoverflow.com/questions/tagged/swiper)
for that.

Before you open an issue, please check if a similar issue already exists or has been closed before.

**Comment by jrgwerum** [comment]
I can confirm that behavior. On my case it`s a bootstrap vue combobox.

**Comment by jrgwerum** [comment]
I`ve figured out that there exists something like a whitelisting for form elements on swiper see https://github.com/nolimits4web/swiper/blob/704624f9c015481d94049cdacf023cb6993c38cc/src/components/core/core-class.js#L225

Bootstrap Vue adds some extra ui elements based on divs or HtmlListNodes and it seems that they must be whitelisted too.

I`ve extended the list with div, ul and li and it seems that everything works now.

```
// Form elements to match
formElements: 'div, ul, li, input, select, option, textarea, button, video, label',
```

**Comment by iBiber** [comment]
We use vue-select on a swiper slide and have similar problems with swiper 6.3.5 in desktop mode.

vue-select is a composed element, which consists of:
* a _span_ for the text of a selected item
* an _input_ text field to change the selection via keyboard
* a _button_ to clear the selection
* a _svg_ for as the arrorw button to display if the selection box is open or collapsed
* a _ul_ / _li_ for the opened selection list

If I add 'div, ul, li' to the list of formElements, then the selection list is more functional (items can be selected from the list), but the _svg_-button and the _span_ for the selected item is still not clickable.

Probably I can add _span_ and _svg_, but I guess there are reasons why Swiper catches the click events. I can't see any problems, but I'm a bit worried that I missed something that doesn't work.

**Comment by iBiber** [comment]
I've prepared a CodePen to show the problem:
https://codepen.io/ibiber/pen/NWRWJVj

**Comment by iBiber** [comment]
I think I found a workaround which does work more smoothly.

It looks like there are two problems (in case of vue-select).
The first problem is that the click events are caught (preventDefault).
And the second problem is that swiper calls blur on the active (form) element.

Check out the workaround comments at [my example on CodePen](https://codepen.io/ibiber/pen/NWRWJVj) to play around with:
```
    const swiper = new Swiper(".swiper-container", {
      // Workaround part 1:
      touchStartPreventDefault: false
    })
    // Workaround part 2:
    swiper.touchEventsData.formElements = 'undefined'
```

Part 1: To handle mouse down and click events on all elements, set the swiper parameter `touchStartPreventDefault: false`.
That will disable this code block: https://github.com/nolimits4web/swiper/blob/9dead9ef4ba5d05adf266deb7e3703ceb199a241/src/components/core/events/onTouchStart.js#L90-L97

Part 2: Set `swiper.touchEventsData.formElements = 'undefined'` to define _nothing_ as formElements. That will disable the code block that calls blur: https://github.com/nolimits4web/swiper/blob/9dead9ef4ba5d05adf266deb7e3703ceb199a241/src/components/core/events/onTouchStart.js#L81-L88

I declared this solution as workaround because I think overwriting the formElements is not an intended use.
